atlp-rwanda/stackup2-barefoot-backend
# #171380754 make multi city request

**What does this PR do?**
For creating a feature of making multiple cities request

**Description of Task to be completed?**

- add src/database/migrations/20200312122638-create-request.js
- add src/database/models/request.js
- add validations for multi city in validations utils
- write multi city logic in createTripRequest controller function

**How should this be manually tested?**

- Clone `ft-make-multi-city-request-171380754` branch
- Open the mentioned branch in cmd, and then write npm install to install packages
- Run the project by writing in cmd npm run dev.
- Open postman and then run signup route to register -> run login route to authenticate
- Set authentication header and try to send post request of trips route `localhost:3000/api/trips/`
- After the mentioned steps, the trip request will be created and returns  success message and data
_`By the way travelFrom, travelTo and TravelDate, each one should be in array form. for more information look on screenshot below.`_

Mock Data

{
    travelFrom: ['Butare', 'Musanze'],
    travelTo: ['Musanze', 'Rusizi'],
    travelDate: ['2020-08-09', '2020-08-16'],
    travelReason: 'business meeting',
    travelType: 'multi-cities',
    accommodation: true
  },
 {
    travelFrom: ['Butare', 'Musanze', 'Kayonza'],
    travelTo: ['Musanze', 'Kayonza', 'Huye'],
    travelDate: ['2020-08-09', '2020-08-16', '2020-08-24'],
    travelReason: 'business meeting',
    travelType: 'multi-cities',
    accommodation: true
  },

**Any background context you want to provide?**
N/A

**What are the relevant pivotal tracker stories?**
[#171380754](https://www.pivotaltracker.com/n/projects/2435482/stories/171380754)

![Capture23](https://user-images.githubusercontent.com/23632307/77783052-e0dd6b00-7060-11ea-9381-dc7f143f985d.PNG)

![Screenshot (2)](https://user-images.githubusercontent.com/23632307/77230911-5e583580-6ba0-11ea-8675-1d5eb591ca2b.png)
